### PR TITLE
Document the role of `missing` in `PartialQuickSort` and deprecate `PartialQuickSort(::Integer)`

### PR DESCRIPTION
**Pull request by LilithHafner**
From https://github.com/JuliaLang/julia/pull/47191#issuecomment-1288068650

### CODE REVIEW

File: base/sort.jl [hunk 1]
```diff
@@ -434,8 +434,8 @@ struct PartialQuickSort{L<:Union{Integer,Missing}, H<:Union{Integer,Missing}} <:
     lo::L
     hi::H
 end
-PartialQuickSort(k::Integer) = PartialQuickSort(missing, k)
 PartialQuickSort(k::OrdinalRange) = PartialQuickSort(first(k), last(k))
+@deprecate PartialQuickSort(k::Integer) PartialQuickSort(missing, k)
```
Comment (petvana) [inline base/sort.jl:438]: Haven't use `@deprecate`, but it seems it should be placed into `base/deprecated.jl`.

**Comment by LilithHafner** [comment]
I also haven't used deprecations in base before, and I'd like approval from someone who is confident that this is acceptable w.r.t. SymVer and our promise of backward compatibility.

**Comment by petvana** [comment]
As CI complains, the docs should be updated after merging your amazing stable `QuickSort` algorithm in #45222

https://github.com/JuliaLang/julia/blame/master/doc/src/base/sort.md#L143-L190

Just a few comments for `sort.md`
- [ ]  `QuickSort` is stable
- [ ] `QuickSort` is not guaranteed to be O(n log n), strictly speaking
- [ ] `PartialQuickSort(k)` needs to be removed
- [ ] `defalg(v::AbstractArray) = MergeSort` is outdated

Further for `sort.jl`
- [ ]  `MergeSort` is never default alg.
https://github.com/JuliaLang/julia/blob/54e68997108d33a6478bc1bfbbc2b7d1ea072202/base/sort.jl#L889-L893

Since I'm writing, I cannot find the stabilization of `QuickSort` in `NEWS.md`, and I think it's worthy.

More general question. Is there a plan to guarantee stability of `sort(x)` in v1.9 on? (And for `sortpertm!`-like functions?) The stability is now guaranteed (by documentation) only for `QuickSort` and `AdaptiveSort`. Thus the only way to rely on the stability is to use `sort(x, alg=QuickSort)` or `sort(x, alg=AdaptiveSort)`, right?

I can try preparing some PRs, if you want.

**Comment by LilithHafner** [comment]
Thank you for your observations! Yes, sort.md needs updating.

> I can try preparing some PRs, if you want.

That would be lovely! Thank you! I think the `PartialQuickSort` entry should be changed, but not removed—it is still a valuable algorithm we have to offer, now even more so than before due to added performance in ranges that do not include the beginning of the vector. Also, the whole defalg system now seems a bit obsolete so the current state may still be in flux. Nevertheless, it is still clearly wrong as is, so a change would be warranted.

I happened to make a PR for the sort.jl problem you noted before I realized you were offering to make PRs, I'd be happy if you took as many of the rest as you'd like to :)

**Comment by LilithHafner** [comment]
This PR deprecates improper usage of PartialQuickSort that is still found in sort.jl, so tests will fail until #47191 which fixes those usages is merged.

**Comment by LilithHafner** [comment]
CI now passes (failures unrelated) and all this needs is approval from someone familiar with adding deprecations to Base (@vtjnash, perhaps)

**Pull request closed without merging**